src/components/GameDetail/GameProgress.vue

```diff
@@ -0,0 +1,423 @@
+      <div
+        class="progress-data"
+        v-if="!pie"
+      >
+        <progress
+          class="progress-bar"
+          max="100"
+          :value="localProgress.number"
+        >
+          <div class="progress-bar-fallback">
+            <span :style="style">
+              Progress: {{localProgress.number}}%
+            </span>
+          </div>
+        </progress>
+        <div
+          v-if="localProgress.number"
+          class="progress-bar-label"
+        >
+          {{localProgress.number}}%
+        </div>
+      </div>
+
+      <div
+        v-else
+        class="progress-pie"
+        :data-value="localProgress.number"
+      >
+      </div>
+
+      <button
+        v-if="!small && !pie"
+        class="primary"
+        @click="editProgress"
+      >
+        Edit progress
+      </button>
+    </div>
+
+    <div
+      class="progress-field"
+      v-if="showProgressField && !small && !pie"
+    >
+      <div class="progress-data">
+        <div
+          v-if="localProgress.number"
+          class="progress-bar-label"
+        >
+          {{localProgress.number}}%
+        </div>
+
+        <input
+          class="progress-range"
+          v-model.number="localProgress.number"
+          placeholder="50"
+          type="range"
+          min="0"
+          max="100"
+          step="5"
+          autofocus
+        >
+      </div>
+
+      <div class="progress-action">
+        <div class="progress-edit">
+          <button class="secondary" @click="reset">
+            {{ $t('global.cancel') }}
+          </button>
+
+          <button class="primary" @click="saveProgress">
+            {{ $t('global.save') }}
+          </button>
+        </div>
+
+        <button class="danger" @click="deleteProgress">
+          Delete progress
+        </button>
+      </div>
+    </div>
+
+    <button
+      v-if="!hasProgress &&
+            !showProgressField &&
+            !small &&
+            !pie"
+      class="primary"
+      @click="addProgress"
+    >
+      <i class="fas fa-clock" />
+      {{ $t('progresses.addProgress') }}
+    </button>
+  </div>
+</template>
+
+<script>
+import { mapState, mapGetters } from 'vuex';
+
+export default {
+  props: {
+    small: {
+      type: Boolean,
+      default: false,
+    },
+    pie: {
+      type: Boolean,
+      default: false,
+    },
+  },
+
+  data() {
+    return {
+      showProgressField: false,
+      localProgress: {
+        number: null,
+      },
+    };
+  },
+
+  computed: {
+    ...mapState(['game', 'progresses']),
+    ...mapGetters(['gameProgress']),
+
+    hasProgress() {
+      return this.gameProgress && this.gameProgress.number;
+    },
+
+    style() {
+      return `width: ${this.localProgress.number}%`;
+    },
+  },
+
+  mounted() {
+    this.reset();
+  },
+
+  methods: {
+    reset() {
+      this.localProgress = this.gameProgress
+        ? JSON.parse(JSON.stringify(this.gameProgress))
+        : { number: null };
```

Comment:
`number` seems too generic, consider renaming `number` to `progress`, `progressPercentage` or simply `value`.